packages/wallet/src/utils/hex-utils.ts

```diff
@@ -16,13 +16,5 @@ export function eqHex(a: string, b: string) {
 }
 
 export function eqHexArray(a: string[], b: string[]): boolean {
-  if (a.length !== b.length) {
-    return false;
-  }
-  for (let i = 0; i < a.length; i++) {
-    if (!eqHex(a[i], b[i])) {
-      return false;
-    }
-  }
-  return true;
+  return a.toString() === b.toString();
```

Comment:
This will return false if two hex numbers have different string representations. IE `['0x01]` and '[0x001]`.